eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageRequestProtocolResolver.java

```diff
@@ -59,81 +59,64 @@ public static CloudEvent buildEvent(Header header, Body body) throws ProtocolHan
-                    .withExtension(ProtocolKey.ClientInstanceKey.IDC, idc)
-                    .withExtension(ProtocolKey.ClientInstanceKey.IP, ip)
-                    .withExtension(ProtocolKey.ClientInstanceKey.PID, pid)
-                    .withExtension(ProtocolKey.ClientInstanceKey.SYS, sys)
-                    .withExtension(ProtocolKey.ClientInstanceKey.USERNAME, username)
-                    .withExtension(ProtocolKey.ClientInstanceKey.PASSWD, passwd)
-                    .withExtension(ProtocolKey.VERSION, version.getVersion())
-                    .withExtension(ProtocolKey.LANGUAGE, language)
-                    .withExtension(ProtocolKey.PROTOCOL_TYPE, protocolType)
-                    .withExtension(ProtocolKey.PROTOCOL_DESC, protocolDesc)
-                    .withExtension(ProtocolKey.PROTOCOL_VERSION, protocolVersion)
-                    .withExtension(SendMessageRequestBody.BIZSEQNO, sendMessageRequestBody.getBizSeqNo())
-                    .withExtension(SendMessageRequestBody.UNIQUEID, sendMessageRequestBody.getUniqueId())
-                    .withExtension(SendMessageRequestBody.PRODUCERGROUP,
-                        sendMessageRequestBody.getProducerGroup())
-                    .withExtension(SendMessageRequestBody.TTL, sendMessageRequestBody.getTtl());
-                if (StringUtils.isNotEmpty(sendMessageRequestBody.getTag())) {
-                    cloudEventBuilder = cloudEventBuilder.withExtension(SendMessageRequestBody.TAG, sendMessageRequestBody.getTag());
-                }
-                if (sendMessageRequestBody.getExtFields() != null && sendMessageRequestBody.getExtFields().size() > 0) {
-                    for (Map.Entry<String, String> entry : sendMessageRequestBody.getExtFields().entrySet()) {
-                        cloudEventBuilder = cloudEventBuilder.withExtension(entry.getKey(), entry.getValue());
-                    }
-                }
-                event = cloudEventBuilder.build();
+            switch (protocolVersion) {
+                case SpecVersion.V1.toString():
+                    cloudEventBuilder = CloudEventBuilder.v1();
+                    break;
+                case SpecVersion.V03.toString():
+                    cloudEventBuilder = CloudEventBuilder.v03();
+                    break;
+                default:
+                    return null; // unsupported protocol version
             }
+
+            CloudEvent event = getBuilderCloudEvent(sendMessageRequestBody, protocolType, protocolDesc, protocolVersion,
+                code, env, idc, ip, pid, sys, username, passwd, version, language, content, cloudEventBuilder);
             return event;
+
         } catch (Exception e) {
             throw new ProtocolHandleException(e.getMessage(), e.getCause());
         }
     }
+
+    private static CloudEvent getBuilderCloudEvent(SendMessageRequestBody sendMessageRequestBody, String protocolType, String protocolDesc,
```

Comment:
There are too many parameters in this method, maybe we can simplify it, I think it can be like this:
```
        private static CloudEvent getCloudBuilderEvent(SendMessageRequestBody sendMessageRequestBody,
                                                   SendMessageRequestHeader sendMessageRequestHeader,
                                                   CloudEventBuilder cloudEventBuilder) {
        final CloudEvent event;
        cloudEventBuilder = cloudEventBuilder.withId(sendMessageRequestBody.getBizSeqNo())
            .withSubject(sendMessageRequestBody.getTopic())
            .withType("eventmeshmessage")
            .withSource(URI.create("/"))
            .withData(sendMessageRequestBody.getContent().getBytes(StandardCharsets.UTF_8))
            .withExtension(ProtocolKey.REQUEST_CODE, sendMessageRequestHeader.getCode())
            .withExtension(ProtocolKey.ClientInstanceKey.ENV, sendMessageRequestHeader.getEnv())
            .withExtension(ProtocolKey.ClientInstanceKey.IDC, sendMessageRequestHeader.getIdc())
            .withExtension(ProtocolKey.ClientInstanceKey.IP, sendMessageRequestHeader.getIp())
            .withExtension(ProtocolKey.ClientInstanceKey.PID, sendMessageRequestHeader.getPid())
            .withExtension(ProtocolKey.ClientInstanceKey.SYS, sendMessageRequestHeader.getSys())
            .withExtension(ProtocolKey.ClientInstanceKey.USERNAME, sendMessageRequestHeader.getUsername())
            .withExtension(ProtocolKey.ClientInstanceKey.PASSWD, sendMessageRequestHeader.getPasswd())
            .withExtension(ProtocolKey.VERSION, sendMessageRequestHeader.getVersion().getVersion())
            .withExtension(ProtocolKey.LANGUAGE, sendMessageRequestHeader.getLanguage())
            .withExtension(ProtocolKey.PROTOCOL_TYPE, sendMessageRequestHeader.getProtocolType())
            .withExtension(ProtocolKey.PROTOCOL_DESC, sendMessageRequestHeader.getProtocolDesc())
            .withExtension(ProtocolKey.PROTOCOL_VERSION, sendMessageRequestHeader.getProtocolVersion())
            .withExtension(SendMessageRequestBody.BIZSEQNO, sendMessageRequestBody.getBizSeqNo())
            .withExtension(SendMessageRequestBody.UNIQUEID, sendMessageRequestBody.getUniqueId())
            .withExtension(SendMessageRequestBody.PRODUCERGROUP,
                sendMessageRequestBody.getProducerGroup())
            .withExtension(SendMessageRequestBody.TTL, sendMessageRequestBody.getTtl());
        if (StringUtils.isNotEmpty(sendMessageRequestBody.getTag())) {
            cloudEventBuilder = cloudEventBuilder.withExtension(SendMessageRequestBody.TAG, sendMessageRequestBody.getTag());
        }
        if (sendMessageRequestBody.getExtFields() != null && sendMessageRequestBody.getExtFields().size() > 0) {
            for (Map.Entry<String, String> entry : sendMessageRequestBody.getExtFields().entrySet()) {
                cloudEventBuilder = cloudEventBuilder.withExtension(entry.getKey(), entry.getValue());
            }
        }
        event = cloudEventBuilder.build();
        return event;
	}
```
What do you think?

Reply:
This makes a lot of sense, will update the PR

Reply:
updated the PR, please have a look
thanks